raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -329,8 +387,9 @@ private void appendLeaderChangeMessage(LeaderState state, long currentTimeMs) {
     }
 
     private void flushLeaderLog(LeaderState state, long currentTimeMs) {
-        log.flush();
+        // We update the end offset before flushing so that parked fetches can return sooner
```

Comment:
Yeah, it's ok for followers to see uncommitted or even unflushed data. The main thing is that we avoid advancing the high watermark until the fsync completes. Note that this is the main reason that we had to do KAFKA-10527. Without this fix, it was possible for the leader to continue in the same epoch after a start, which means that it could lose and overwrite unflushed data.